gordo_components/dataset/base.py

```diff
@@ -111,7 +124,24 @@ def join_timeseries(
             logging.debug(series.head(3))
             logging.debug(series.tail(3))
 
-            resampled = series.resample(resolution, label="left").mean()
+            resampled = series.resample(resolution, label="left").agg(
+                aggregation_methods
+            )
+            # If several aggregation methods are provided, agg returns a dataframe
+            # instead of a series. In this dataframe the column names are the
+            # aggregation methods, like "max" and "mean", so we have to make a
+            # multi-index with the series-name as the top-level and the
+            # aggregation-method as the lower-level index
+            if isinstance(
+                resampled, pd.DataFrame
+            ):  # Several aggregation methods provided
```

Comment:
Nit-pick alert: Could save a couple lines by putting this comment either above or below the `if`